ddtrace/appsec/_iast/_stacktrace.c

```diff
@@ -39,68 +51,82 @@
-        frame = GET_FRAME(tstate);
-        while (NULL != frame) {
-            filename_o = GET_FILENAME(frame);
-            const char* filename = PyUnicode_AsUTF8(filename_o);
-            if (((strstr(filename, DD_TRACE_INSTALLED_PREFIX) != NULL && strstr(filename, TESTS_PREFIX) == NULL)) ||
-                (strstr(filename, SITE_PACKAGES_PREFIX) != NULL || strstr(filename, cwd) == NULL)) {
+    PyFrameObject* frame = GET_FRAME(tstate);
+    if (!frame) {
+        Py_DECREF(cwd_bytes);
+        return NULL;
+    }
 
-                frame = GET_PREVIOUS(frame);
-                continue;
-            }
-            /*
-             frame->f_lineno will not always return the correct line number
-             you need to call PyCode_Addr2Line().
-            */
-            line = GET_LINENO(frame);
-            return PyTuple_Pack(2, filename_o, Py_BuildValue("i", line));
+    while (NULL != frame) {
+        filename_o = GET_FILENAME(frame);
+        if (!filename_o) {
+            goto exit;
+        }
+        const char* filename = PyUnicode_AsUTF8(filename_o);
+        if (((strstr(filename, DD_TRACE_INSTALLED_PREFIX) != NULL && strstr(filename, TESTS_PREFIX) == NULL)) ||
+            (strstr(filename, SITE_PACKAGES_PREFIX) != NULL || strstr(filename, cwd) == NULL)) {
+            PyFrameObject* prev_frame = GET_PREVIOUS(frame);
+            FRAME_DECREF(frame);
+            FILENAME_DECREF(filename_o);
+            frame = prev_frame;
+            continue;
         }
+        /*
+         frame->f_lineno will not always return the correct line number
+         you need to call PyCode_Addr2Line().
+        */
+        line = GET_LINENO(frame);
+        PyObject* line_obj = Py_BuildValue("i", line);
+        if (!line_obj) {
+            goto exit;
+        }
+        result = PyTuple_Pack(2, filename_o, line_obj);
+        break;
     }
-#if PY_MAJOR_VERSION > 3 || PY_MAJOR_VERSION == 3 && PY_MINOR_VERSION >= 10
-    return Py_NewRef(Py_None);
-#else
-    Py_INCREF(Py_None);
-    return Py_None;
-#endif
+exit:
+    Py_DECREF(cwd_bytes);
+    FRAME_DECREF(frame);
+    FILENAME_XDECREF(filename_o);
+    return result;
 }
 
 static PyMethodDef StacktraceMethods[] = {
-    { "get_info_frame", (PyCFunction)get_file_and_line, METH_VARARGS, "stacktrace functions" },
+    { "get_info_frame", (PyCFunction)get_file_and_line, METH_O, "stacktrace functions" },
     { NULL, NULL, 0, NULL }
 };
 
 static struct PyModuleDef stacktrace = { PyModuleDef_HEAD_INIT,
                                          "ddtrace.appsec._iast._stacktrace",
                                          "stacktrace module",
                                          -1,
-                                         StacktraceMethods };
+                                         StacktraceMethods};
 
 PyMODINIT_FUNC
 PyInit__stacktrace(void)
 {
-    PyObject* m;
-    m = PyModule_Create(&stacktrace);
+    PyObject* m = PyModule_Create(&stacktrace);
```

Comment:
unrelated change?

Reply:
Technically is unrelated but I added it as cleanup s I was already changing most of the code. I can revert if you want.

Reply:
You need to run this before committing your changes:

```bash
clang-format --style="{BasedOnStyle: Mozilla, IndentWidth: 4, ColumnLimit: 120}" -i ddtrace/appsec/_iast/_stacktrace.c
```

Alternatively, you can use pre-commit 🙃

Reply:
Fixed!